src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  Object.values(items).forEach((itemKey) => {
+    if (String(itemKey[key])) {
```

Comment:
dry
move this to the separate variable
```suggestion
    if (...) {
```